frappe/integrations/utils.py

```diff
@@ -20,14 +19,15 @@ def make_request(method: str, url: str, auth=None, headers=None, data=None, json
 		response = frappe.flags.integration_request = s.request(
 			method, url, data=data, auth=auth, headers=headers, json=json, params=params
 		)
-		response.raise_for_status()
-
-		if response.headers.get("content-type") == "text/plain; charset=utf-8":
+		content_type = response.headers.get("content-type")
+		if content_type == "text/plain; charset=utf-8":
 			return parse_qs(response.text)
-		elif not response.text:
-			return
+		elif content_type == "application/json":
```

Comment:
There are types of json too. `ct.startswith("application/") and ct.endswith("+json")`

Reply:
Also charset can be mentioned, which will cause this to fail